pkg/model/components/openstack.go

```diff
@@ -32,30 +32,27 @@ var _ loader.OptionsBuilder = &OpenStackOptionsBulder{}
 func (b *OpenStackOptionsBulder) BuildOptions(o interface{}) error {
 	clusterSpec := o.(*kops.ClusterSpec)
 
-	if clusterSpec.GetCloudProvider() != kops.CloudProviderOpenstack {
+	if clusterSpec.CloudProvider.Openstack == nil {
 		return nil
 	}
 
 	if clusterSpec.CloudConfig == nil {
 		clusterSpec.CloudConfig = &kops.CloudConfiguration{}
 	}
 
-	if clusterSpec.CloudConfig.Openstack == nil {
-		clusterSpec.CloudConfig.Openstack = &kops.OpenstackConfiguration{}
-	}
-	if clusterSpec.CloudConfig.Openstack.BlockStorage == nil {
-		clusterSpec.CloudConfig.Openstack.BlockStorage = &kops.OpenstackBlockStorageConfig{}
+	if clusterSpec.CloudProvider.Openstack.BlockStorage == nil {
```

Comment:
It might be easier to have a field `config` that points to clusterSpec.CloudProvider.Openstack, but I recognize that this is pre-existing.